### ISSUE TITLE
Error in rank hinge loss

### ISSUE THREAD

**Issue by aneesh-joshi** [faneshion/MatchZoo#228]
https://github.com/faneshion/MatchZoo/blob/50422a5bc973d27b8c507122aa3f41e633366893/matchzoo/losses/rank_losses.py#L16

I am not an expert but this seems to be wrong.
All loss functions use `y_true` and `y_pred`
This function only uses `y_pred`

References:
https://github.com/keras-team/keras/blob/468f080c98f06780c950e5a78c9eeeaf9fff002e/keras/losses.py#L38

@uduse @bwanglzu @yangliuy

**Comment by aneesh-joshi** [comment]
Oh
just realized MZ has all `y_true` as alternate 1 and 0
so, here the assumption about `y_true` has been made.

Maybe this should change in `2.0`?

Will close the issue after discussion.

**Comment by bwanglzu** [comment]
I think the code is correct, but the implementation is kind of not easy to read, we'll clean it in 2.0, thanks! @aneesh-joshi

**Comment by aneesh-joshi** [comment]
@bwanglzu The implementation is correct.
I misunderstood due to the underlying assumption that the correct and wrong options alternate.

Should I close the issue?

**Comment by bwanglzu** [comment]
@aneesh-joshi please, thanks :+1:
